Long Range Example Requires MPI even when MPI is disabled

The spme example depends on particles.h, which includes mpi. It's not conditional on MPI like the `directsum_example`

https://github.com/ECP-copa/Cabana/blob/500326520268bcaa05e77ca8c9a6a7cb7320230b/core/example/longrange/CMakeLists.txt#L10

building on OSX 10.15 with gcc-9 from homebrew, kokkos 3 from custom cmake and FFTW from spack (not linking to MPI):

FFTW=/usr/local/spack/opt/spack/darwin-catalina-ivybridge/gcc-9/fftw-3.3.8-bqqkr
cu7l6u6sghivfd2zxlqw2oworn3
cmake .. \
  -D CMAKE_CXX_COMPILER=`which g++-9` \
  -D CMAKE_BUILD_TYPE="Release" \
  -D Kokkos_DIR=/usr/local/lib/cmake/Kokkos \
  -D CMAKE_PREFIX_PATH=/usr/local \
  -D CMAKE_INSTALL_PREFIX=/usr/local \
  -D Cabana_ENABLE_Serial=OFF \
  -D Cabana_ENABLE_OpenMP=ON \
  -D Cabana_ENABLE_TESTING=ON \
  -D Cabana_ENABLE_EXAMPLES=ON \
  -D FFTW3_INCLUDE_DIR="$FFTW/include" \
  -D FFTW3_LIBRARY="$FFTW/lib/libfftw3.3.dylib"

<pre>
[ 98%] Building CXX object core/example/longrange/CMakeFiles/SPME.dir/spme_example.cpp.o
In file included from /Users/rogers/build/Cabana/core/example/longrange/spme_example.cpp:2:
/Users/rogers/build/Cabana/core/example/longrange/particles.cpp:15:10: fatal error: mpi.h: No such file or directory
   15 | #include "mpi.h"
      |          ^~~~~~~
compilation terminated.
make[2]: *** [core/example/longrange/CMakeFiles/SPME.dir/spme_example.cpp.o] Error 1
make[1]: *** [core/example/longrange/CMakeFiles/SPME.dir/all] Error 2
make: *** [all] Error 2
</pre>

--- a/core/example/longrange/CMakeLists.txt
+++ b/core/example/longrange/CMakeLists.txt
@@ -1,9 +1,11 @@
-if(Cabana_ENABLE_MPI)
-  add_executable(Direct directsum_example.cpp)
-  target_link_libraries(Direct cabanacore)
-  add_executable(Ewald ewaldsum_example.cpp)
-  target_link_libraries(Ewald cabanacore)
+if(NOT Cabana_ENABLE_MPI)
+  return()
 endif()
+  
+add_executable(Direct directsum_example.cpp)
+target_link_libraries(Direct cabanacore)
+add_executable(Ewald ewaldsum_example.cpp)
+target_link_libraries(Ewald cabanacore)
 
 if(Cabana_ENABLE_Cuda)
   find_package(CUDA REQUIRED)

